x-pack/plugin/ql/src/main/java/org/elasticsearch/xpack/ql/util/Check.java

```diff
@@ -47,4 +49,13 @@ public static void isBoolean(Object obj) {
             throw new QlIllegalArgumentException("A boolean is required; received [{}]", obj);
         }
     }
+
+    // 18446744073709551615
+    public static final BigInteger UNSIGNED_LONG_MAX = BigInteger.ONE.shiftLeft(Long.SIZE).subtract(BigInteger.ONE);
+
+    public static void isUnsignedLong(BigInteger bi) {
```

Comment:
> There's a slightly different out of range check here

The signum is a pre-calculated value in BigInteger and readily available and so the comparison against `.ZERO` costlier.

> Also, I am not sure this is the right place for this kind of check.

Fair enough. I've taken Mario's [suggestion](https://github.com/elastic/elasticsearch/pull/65145#discussion_r528770855) and put it in a utility class. I do believe this check should have it's own function, rather than inlined, since it's the very definition of the "unsigned long" (i.e. an arbitrary range limitation of a BigInteger, in our case). This function could be placed in respective user classes too, but I'd personally avoid the duplication. Anyways, open for improvement suggestions.